tycho-targetplatform/src/main/java/org/eclipse/tycho/targetplatform/TargetDefinitionFile.java

```diff
@@ -563,6 +563,10 @@ public static boolean isTargetFile(File file) {
 	            && !file.getName().startsWith(".polyglot.");
 	}
 
+	public static TargetDefinition.Repository repository(String id, String uri) {
```

Comment:
Why is this required and one can not just call the constructor directly?

Reply:
I found this less intrusive than making the private type public, but if you have no problem with that I'll just increase visibility on the Repository class and remove this

Reply:
But where is this method actually used?

Reply:
in `UpdateTargetMojo.LatestVersionLocation#getRepositories` when creating copy of repository with its location resolved

Reply:
In this case it would be better to have a private implementation in `UpdateTargetMojo` that can delegate everything to original item but return a different URL ...